packages/jdk.rb

```diff
@@ -1,43 +1,43 @@
 require 'package'
 
 class Jdk < Package
-  description 'The JDK is a development environment for building applications, applets, and components using the Java programming language.'
-  homepage 'https://www.oracle.com/java/technologies/javase/javase-jdk8-downloads.html'
-  version '8u333-18.0.1.1'
+  description 'The Oracle JDK is a development environment for building applications, applets, and components using the Java programming language.'
+  homepage 'https://www.oracle.com/java'
+  version 'LATEST'
   license 'Oracle-BCLA-JavaSE'
   compatibility 'all'
 
   is_fake
 
   def self.preflight
-    jdkver = `java -version 2>&1 | head -1 | cut -d'"' -f2`.chomp
-    unless jdkver.empty? || jdkver.include?('No such file or directory') || jdkver.include?('not found')
-      majver = `java -version 2>&1 | head -1 | cut -d'"' -f2 | cut -d'.' -f1`.chomp
-      majver = '8' if majver == '1'
-      puts "Package jdk#{majver} already installed.".lightgreen
-      abort "Enter `crew remove jdk#{majver} && crew install jdk` to install a different version."
-    end
-    puts
-    puts 'Select version:'
-    puts ' 8 = JDK 8u333'
-    if ARCH == 'x86_64'
-      puts '11 = JDK 11.0.16.1'
-      puts '17 = JDK 17.0.3.1'
-      puts '18 = JDK 18.0.1.1'
-    end
-    puts ' 0 = Cancel'
-
-    while version = $stdin.gets.chomp
-      case version
-      when '8', '11', '17', '18'
-        depends_on "jdk#{version}"
-        break
-      when '0'
-        abort
-        break
-      else
-        puts "\nPlease select from one of the options or enter 0 to cancel."
+    if ARGV.include?('install')
+      if File.exist?("#{CREW_PREFIX}/bin/java")
+        jdkver_str = `#{CREW_PREFIX}/bin/java -version 2>&1`
+        is_openjdk = jdkver_str.include?('openjdk')
+        jdkver     = jdkver_str[/version "(.+?)"/, 1]
+        majver     = jdkver.split('.')[0]
+        majver     = '8' if majver == '1'
+        pkg_suffix = is_openjdk ? 'openjdk' : 'jdk'
+
+        puts "Package #{pkg_suffix}#{majver} already installed.".lightgreen
+        warn "Run `crew remove #{pkg_suffix}#{majver} && crew install jdk` to install a different version.".yellow
+        return false
+      end
+
+      jdk_latest_ver = %w[8 11 17 18].to_h do |jdk_majver|
```

Comment:
How about instead of hardcoding the versions, search for `jdk#.rb` in packages and use whatever is found instead?  That way never packages are added without modifying the code.

Reply:
That's what this block initially do :)